### PR TITLE
SSS: Parameter indexing

### PR DESCRIPTION
**Pull request by smowton**
A simple utility for the common task of mapping parameter indices onto their indices. Seems like it'd be generally useful.

### CODE REVIEW

File: src/util/parameter_indices.cpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Author: Chris Smowton, chris.smowton@diffblue.com
+
+\*******************************************************************/
+
+#include "parameter_indices.h"
+
+std::map<irep_idt, std::size_t> get_parameter_indices(const code_typet &fn_type)
+{
+  std::map<irep_idt, std::size_t>  parameter_indices;
+  for(std::size_t i = 0; i != fn_type.parameters().size(); ++i)
```
Comment (peterschrammel) [inline src/util/parameter_indices.cpp]: cpplint

File: src/util/parameter_indices.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+Module: Parameter indexing
+
+Author: Chris Smowton, chris.smowton@diffblue.com
+
+\*******************************************************************/
+
+#include "parameter_indices.h"
+
+std::map<irep_idt, std::size_t> get_parameter_indices(const code_typet &fn_type)
```
Comment (peterschrammel) [inline src/util/parameter_indices.cpp]: function comment block (also explain the rationale where this function might be useful)

**Comment by smowton** [comment]
done

**Comment by tautschnig** [comment]
Again, apologies for being late: would someone mind pointing out the use case of such code? Is meant for name-based lookups?

**Comment by smowton** [comment]
Yes. IIRC I had an array of parameter-related things and some assignments to them by name, but the specifics beyond that escape me for now.

**Comment by tautschnig** [comment]
Understood; might be nice to have sufficient context in a single PR. In some distant future it will be really hard for anyone to understand why feature xyz has been added.

**Pull request merged**